### DISCUSSION TITLE
[ FF7 ] Direct3D9 sprite transparency rendering glitch

### DISCUSSION THREAD

**Discussion by fscanp** [julianxhokaxhiu/FFNx#335] [Q&A] [answered]
Using Direct3D9 will result in what seems to be a sprite transparency rendering glitch. Everything else seems to work just fine. I understand that d3d9 rendering is labeled as a tech review and is not recommended to be used, but since I installed this ff7 on my old pc with old GeForce 8600gt card that does not fully support d3d11, I have no other options as that rendering is the only one that works on this card. Well either that or just stop using mods all together. 

I'm using '98 version of FF7 with whatever version of FFNx is shipped with 7th Heaven 2.3.3. The top lines of the log is: 
[00000000] INFO: FFNx driver version 1.9.0.208
[00000000] TRACE: v1: 0x99CE0805, v2: 0xAC0D8B00
so I'm guessing that's the version of FFNx they're bundling. I see that the latest version is 1.10 so if you say it's already fixed in that version, please just ignore this and let me know where I can try to update manuallyto that version

![image](https://user-images.githubusercontent.com/36881244/147592559-c7a64890-092b-403d-a5c5-44a5dc65577d.png)

**Comment by eve-atum** [comment, marked as answer]
The "tech review" aspect of D3D9 is because the graphics library FFNx uses, bgfx, has some bugs in that pipeline.  Since fixing D9 requires the upstream to fix, you might have better luck switching to OpenGL instead.

**Reply by fscanp** [reply]
Yeah I figured. Unfortunately (for me) this old vga card wont even run the game on OpenGl.. I would hear the first seconds of the eidos opening vid (with blank screen), and then it stops, restarts again after a few seconds, repeat ad nauseam. It's probably because the card only supports up to OpenGL 2.0.

**Reply by julianxhokaxhiu** [reply]
I'm sorry about that but FFNx targets modern hardware. DirectX 9 is a leftover for some technical tests under Virtual Machines but far from being ready for general usage. Because of this the only thing I can suggest is migrate to a new hardware, Intel, Nvidia or AMD.

**Reply by fscanp** [reply]
Lol no worries. I installed this on old pc just to introduce my kid to final fantasy franchise story telling game style and thought it might be more enticing if it has more detailed 3d models than the original blocky/chibi style. Just didn't expect that a soon to be 24 years old game to require dx11 hardware. I only checked the min system reqs on the 7th heaven page after seeing the rendering options

**Reply by julianxhokaxhiu** [reply]
Thing is, the game itself doesn't require DX11, but FFNx acting as a rendering engine and modding platform, does. As mentioned, reason is because to enable all the enhancements you can do through mods made by the community, we require a modern stack to achieve those results ( and also, to enable the game to run no modern hardware the right way ).

Weird though you can't run it using OpenGL, what error does it return? Are you using Windows or Linux? One thing you can try to do is run the game on Windows 7.

**Reply by fscanp** [reply]
OpenGL has always been wonky with this vga card for some reason. Sometimes it just works, most of the time it wont. No explanations given. Maybe nvidia couldnt care less about supporting openGL on this card. I already installed the "latest" drivers available. 

For FFNx OpenGL wont display anything. Just blank. I can hear the music of the eidos logo movie playing for a second or two then it stops, Then after some seconds later, it must have restarted as i hear the same thing for another second or two before stopping and restarting again. I wish I could retry it and give you the log file but thing is i already uninstalled 7h and use reunion instead, which works without any problems I should say, even with the high res field and world textures. My kids are enjoying it so I consider it done. Even I must admit I enjoyed it myself after all these years. That and the fact that they play on the old pc so I get to use the newer ones more now that they aren't using it much at least until they finish ff7 and 8 lol.

**Reply by julianxhokaxhiu** [reply]
Glad to hear you found anyway a way out and you're enjoying it. Nevertheless, if you ever find the willingness to test make sure you enable those flags in `FFNx.toml` and then, once you run the game, please post the `FFNx.log` file here so I can double check what is going on and help you find a way to run it ( or maybe fix something, if it has to be fixed ):
```toml
renderer_backend = 1
renderer_debug = true
```

**Reply by fscanp** [reply]
Here you go. I did not load any mods with 7h, just clicked on play. The same thing happens even when I choose the option of play without mods. I waited until it restarts 3 times before killing the process. I hope the debug log tells you something. I'm not gonna uninstall 7h yet just in case you need me to try anything else so just let me know if you do. Cheers!
[FFNx.log](https://github.com/julianxhokaxhiu/FFNx/files/7789439/FFNx.log)

**Reply by julianxhokaxhiu** [reply]
I noticed you're using an old FFNx version. Please update to the latest Stable and try again.

//EDIT: Regarding the shader error, I found you have an issue in common with another user who reported the same and there's a way to fix it but it won't be immediate. Please keep the installation and I'll ping you back again as soon as we have the fix in place to try again. Updating to the latest Stable won't help you atm.

**Reply by fscanp** [reply]
Ahh ok. That's the version that comes with 7th heaven installer and I forgot to update it before clicking play and posting the log file lol. Well I already just got done updating to 1.10.1.0 so it'll be ready whenever you want me to try the fix.

**Reply by julianxhokaxhiu** [reply]
Can you please try the latest canary? Make sure to keep the aforementioned flags enabled and if you notice anything weird, please post again the `FFNx.log` file. Thanks!

**Reply by fscanp** [reply]
Latest canary I see in the releases page v1.10.1.69 and sadly still the same problem. I attached the log file here. If there's a newer build then pls give me the link here. Cheers!

[FFNx.log](https://github.com/julianxhokaxhiu/FFNx/files/7796155/FFNx.log)

**Reply by julianxhokaxhiu** [reply]
Thanks for reporting back. The log file doesn't contain anything useful but perhaps because you forgot to enable these flags on the `FFNx.toml` file:
```toml
renderer_backend = 1
renderer_debug = true
```
Please enable them and try again, and post the log file afterwards.

**Reply by fscanp** [reply]
that's weird. my ffnx.toml already hv renderer_backend set to 1 and renderer_debug set to true. here's the file (renamed to txt so I can attach it here) just in case I'm missing something?

[FFNx.txt](https://github.com/julianxhokaxhiu/FFNx/files/7796180/FFNx.txt)

**Reply by julianxhokaxhiu** [reply]
Config looks good, If you're using 7th make sure you're editing the config file without 7th open otherwise any change will not have an impact. I tested it here locally and it outputs the log as intended ( as your last valid log file )

**Reply by fscanp** [reply]
Lol ok it gets even weirder then. For some reason it just keep making the same useless log file no matter what I try. I've tried changing the renderer option in 7h, close 7h, change it back to 1 in notepad, save, run, and the same useless log file is created each and every time. To make sure, I set the renderer back to dx9 (both in 7h, and then manually in .toml) and ran it (with the same transparency glitch), before then setting it back to openGL (again, both in 7h, and then manually in .toml file). All while checking and deleting the created log file before retrying. The same useless log entries each time. 

A bit more info: I tried running the ff7 exe without 7h and reunion but now it just kept asking me to insert disc 1/2/3 even though the CD is already in the drive (yes it's a physical CD that I managed to keep for 20+ years lol and not iso image file). Maybe it's because i've copied the movie files to hdd and one of the mod manager made the registry changes enough for it not to be able to find my real drive. But this unfortunately also means I cant launch the exe without 7h (or reunion for that matter) to test ffnx

**Reply by julianxhokaxhiu** [reply]
Well, I'm not sure how to help in here.

Regarding Reunion: FFNx is not compatible with Reunion so please make sure you test FFNx on top of a vanilla ff7 installation which has not been patched by anything else on top.

About 7th: If applying changes to the toml file ( for eg. changing renderer backend ) takes effect, then just make sure the other flag is enabled as well. Once you run the game check the log file and see if it's bigger than a few KB ( and it should print WAY MORE log ).

Not sure what is going on and why it worked fine before and why now is not.

//EDIT: Sorry, I just checked the latest canary and indeed I'm not able to output GPU debug info as well. Let me double check what is going on, I tested the functionality only on my own build version I use to check the code quality. I'll ping you again as soon as i fix it.

//EDIT2: Can you please approach me on Discord? Thank you.
